src/node-lib/curriculum-api-2023/queries/subjectListing/processLesson.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  subjects,
+  subjectSlugs,
+  SyntheticUnitvariantLessons,
+} from "@oaknational/oak-curriculum-schema";
+
+interface UnprocessedSubject {
+  [key: string]: {
+    subjectTitle: typeof subjects | string;
+    subjectSlug: typeof subjectSlugs | string;
+    programmeSlug: string;
+    unitIds: Set<number>;
+    lessonSlugs: Set<string>;
+    programmeSlugs: Set<string>;
+  };
+}
+
+interface ProcessedSubject {
+  subjectTitle: typeof subjects | string;
+  subjectSlug: typeof subjectSlugs | string;
+  programmeSlug: string;
+  unitCount: number;
+  lessonCount: number;
+  programmeCount: number;
+}
+
+export const processLessons = (
```

Comment:
a more descriptive name could be useful here, something like `constructSubjectsFromLessonData` ?